Wrappers/Python/cil/processors/Masker.py

```diff
@@ -0,0 +1,168 @@
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='mean', mask=mask, axis=axis)
+
+        return processor
+    
+    @staticmethod
+    def median(mask=None, axis=None):
+        r'''This imputes median where mask == 0. If no axis is specified then operates over flattened array.
+        :param mask: mask 
+        :type mask: DataContainer
+        :param axis: specify axis as int or from 'dimension_labels' to calculate median. 
+        :type axis: str, int
+        '''
+
+        processor = Masker(mode='median', mask=mask, axis=axis)
+
+        return processor
+
+    def __init__(self,
+                 mask = None,
+                 mode = 'value',
+                 value = 0,
+                 axis = None ):
+        
+        r'''Processor to fill missing values provided by mask.
+        :param mask: DataContainer containing a boolean array with zeros where outliers are detected
+        :type mask: DataContainer
+        :param mode: a method to fill in missing values (value, mean, median)
+        :type mode: str, default=value
+        :param value: substitute all outliers with a specific value
+        :type value: float, default=0
+        :param axis: specify axis as int or from 'dimension_labels' to calculate mean or median in respective modes 
+        :type axis: str or int
+        :return: DataContainer or it's subclass with masked outliers
+        :rtype: DataContainer or it's subclass   
+        '''
+
+        kwargs = {'mask': mask,
+                  'mode': mode,
+                  'value': value,
+                  'axis': axis}
+
+        super(Masker, self).__init__(**kwargs)
+    
+    def check_input(self, data):
+
+        if self.mask is None:
+            raise ValueError('Please, provide a mask.')
+
+        if not (isinstance(self.mask, DataContainer)):
+            raise TypeError('Mask must be a DataContainer')
+        
+        if not (issubclass(type(data), DataContainer)):
+            raise TypeError('Processor supports only following data types:\n' +
+                            ' - ImageData\n - AcquisitionData\n' +
+                            ' - DataContainer')
+
+        if not (data.shape == self.mask.shape and data.dimension_labels == self.mask.dimension_labels):
+            raise Exception("Mask and Data has to have the same shape and dimension labels." + 
+                            "{} != {} or {} != {}".format(data.shape, self.mask.shape, data.dimension_labels, self.mask.dimension_labels))
+
+        if self.mode not in ['value', 'mean', 'median']:
+            raise Exception("Wrong mode. One of the following is expected:\n" + \
+                            "value, mean, median, interpolation")
+    
+        return True 
+
+    def process(self, out=None):
+        
+        data = self.get_input()
+        arr = data.copy().as_array()
+        mask_arr = self.mask.as_array()
```

Comment:
I think we could accept masks as numpy array inputs too if they are hand created.  All that needs is:
```
        try:
            mask_arr = self.mask.as_array()
        except:
            mask_arr.array
```